src/tests/Search/SearchResultsList.test.tsx

```diff
@@ -0,0 +1,107 @@
+    const { testData } = getTestData();
+    global.fetch = jest.fn(() =>
+      Promise.resolve({
+        json: () => ({
+          results: testData,
+        }),
+      }),
+    ) as jest.Mock;
+    jest.spyOn(global, 'fetch');
+    // set delay to null to prevent test time-out due to useFakeTimers
+    const user = userEvent.setup({ delay: null });
+
+    render(<SearchView />);
+    // focus input to show results
+    const searchInput = screen.getByRole('textbox');
+    await user.click(searchInput);
+
+    const fleshWound = await screen.findByText(
+      "spam - it's just a flesh wound",
+    );
+
+    await user.click(fleshWound);
+    expect(
+      screen.getByTestId('checkbox-2').classList.contains('Mui-checked'),
+    ).toBe(true);
+  });
+
+  it('should clear the checkbox when a checked result is clicked', async () => {
+    const { testData } = getTestData();
+    global.fetch = jest.fn(() =>
+      Promise.resolve({
+        json: () => ({
+          results: testData,
+        }),
+      }),
+    ) as jest.Mock;
+    jest.spyOn(global, 'fetch');
+    // set delay to null to prevent test time-out due to useFakeTimers
+    const user = userEvent.setup({ delay: null });
+
+    render(<SearchView />);
+    // focus input to show results
+    const searchInput = screen.getByRole('textbox');
+    await user.click(searchInput);
+
+    const fleshWound = await screen.findByText(
+      "spam - it's just a flesh wound",
+    );
+
+    await user.click(fleshWound);
+    expect(
+      screen.getByTestId('checkbox-2').classList.contains('Mui-checked'),
+    ).toBe(true);
+    await user.click(fleshWound);
+    expect(
+      screen.getByTestId('checkbox-2').classList.contains('Mui-checked'),
+    ).toBe(false);
+  });
+
+  it('should not allow selecting more than four revisions', async () => {
+    const { testData } = getTestData();
+    global.fetch = jest.fn(() =>
+      Promise.resolve({
+        json: () => ({
+          results: testData,
+        }),
+      }),
+    ) as jest.Mock;
+    jest.spyOn(global, 'fetch');
+    const spyOnConsole = jest.spyOn(console, 'log');
+    // set delay to null to prevent test time-out due to useFakeTimers
+    const user = userEvent.setup({ delay: null });
+
+    render(<SearchView />);
+    // focus input to show results
+    const searchInput = screen.getByRole('textbox');
+    await user.click(searchInput);
+
+    await user.click(screen.getByTestId('checkbox-1'));
+    await user.click(screen.getByTestId('checkbox-2'));
+    await user.click(screen.getByTestId('checkbox-3'));
+    await user.click(screen.getByTestId('checkbox-4'));
+    await user.click(screen.getByTestId('checkbox-5'));
+
+    expect(
+      screen.getByTestId('checkbox-1').classList.contains('Mui-checked'),
+    ).toBe(true);
+    expect(
+      screen.getByTestId('checkbox-5').classList.contains('Mui-checked'),
+    ).toBe(false);
+
+    expect(spyOnConsole).toHaveBeenCalledWith('maximum four revisions');
```

Comment:
Instead of displaying and testing the message from the console.log(), you could use (either in this PR or another one) the MaterialUI's [Alert ](https://mui.com/material-ui/react-alert/) component to display this message, which should be fairly easy to add.